src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -32,12 +37,16 @@ public OpenFeatureAPI() {
      * @return The singleton instance.
      */
     public static OpenFeatureAPI getInstance() {
+        if (api != null) {
+            return api;
+        }
+
```

Comment:
This change improves performance by only leveraging `synchronized` when we actually create the API instance, instead of EVERY call. Note that we have to check if the api is null again once we've entered the block, since another thread may have already entered and exited the block.